db/migrate/20181203003146_create_users.rb

```diff
@@ -0,0 +1,17 @@
+class CreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :users do |t|
+      t.string   :name
+      t.string   :email
+      t.string   :encrypted_password
+      t.string   :battle_tag
+      t.string   :hotslogs_id
+      t.string   :site_role
```

Comment:
Does this handle multiple roles? Casters can be players, for instance.

Reply:
Yeah, forgot to add that table. I intended `site_role` to delineate between admins and normal users.

Users could have multiple general roles in a separate class and relation.

Reply:
This brings up a good discussion on authorization.

I agree in that we should try to keep authentication the same for everyone (e.g. not need a player who is also a caster have 2 roles assigned to them). We can build a light permissions system that allows certain actions (like requesting to cast) should that person be assigned a particular role.

Reply:
Do we still want `site_role`? I was thinking that admins would have their own place to login, maybe at `/login` from where they can control the levers. What would an admin level user be able to do through the login-walled front-end that a normal user couldn't do?

Reply:
Yeah, I hadn't asked about auth, but I was thinking devise + pundit.

Reply:
As for the `user.site_role` vs `user.roles`, I had envisioned that site_role would be in ["super_admin", "admin", "user"] or how many layers deep we'd need.

Roles would be generic. This is where we could grant things such as:

- `player`
- `captain`
- `caster`
- `other-really-cool-role`